WordPress/openverse
# Upgrade psycopg to version 3 in the API

## Fixes


Related to #2843. The project to convert the Django app to an ASGI application.

## Description


Removes the dependency on `psycopg2` in favor of psycopg 3. [Django 4.2 introduced support for psycopg 3](https://docs.djangoproject.com/en/4.2/releases/4.2/#psycopg-3-support) and the notes says  that _Support for psycopg2 is likely to be deprecated and removed at some point in the future_. Psycopg 3 allows the use of asyncio-based concurrency and static typing, among other improvements.

You can see more of the differences in the [official docs](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html).

## Testing Instructions

Run the project and see around that everything is working as usual.
```
just api/recreate
```

Also, confirms that `pgcli` continues to work:
```
just api/pgcli
```

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
